helper/restore_helper.go

```diff
@@ -507,12 +552,28 @@ func getSubsetFlag(fileToRead string, pluginConfig *utils.PluginConfig) bool {
 	return true
 }
 
-func startRestorePluginCommand(fileToRead string, objToc *toc.SegmentTOC, oidList []int, errBuffer *bytes.Buffer) (io.Reader, bool, error) {
+// pluginCmd is needed to keep track of readable stderr and whether the command
+// has already been finalized.
+type pluginCmd struct {
+	*exec.Cmd
+	stderrBuffer *bytes.Buffer
+	isEnded      bool
+}
+
+func newPluginCmd(errBuf *bytes.Buffer, name string, arg ...string) *pluginCmd {
+	cmd := exec.Command(name, arg...)
+	cmd.Stderr = errBuf
+	return &pluginCmd{cmd, errBuf, false}
+}
+
+func startRestorePluginCommand(fileToRead string, objToc *toc.SegmentTOC, oidList []int) (*pluginCmd, io.Reader, bool, error) {
+	var errBuf bytes.Buffer
```

Comment:
Is it ok that we reuse one buffer for stderr again?

Reply:
We still use separate buffers, the member is moved from `reader.errBuf` to `reader.pluginCmd.stderrBuffer`.

Reply:
Indeed, messed up a bit:)

But why not to move `var errBuf bytes.Buffer` into `newPluginCmd()`?